xbmc/pvr/epg/EpgInfoTag.cpp

```diff
@@ -418,8 +398,32 @@ int CPVREpgInfoTag::GenreSubType() const
   return m_iGenreSubType;
 }
 
+std::string CPVREpgInfoTag::GenreDescription() const
+{
+  return m_strGenreDescription;
+}
+
 const std::vector<std::string> CPVREpgInfoTag::Genre() const
 {
+  if (m_genre.empty())
+  {
+    if ((m_iGenreType == EPG_GENRE_USE_STRING || m_iGenreSubType == EPG_GENRE_USE_STRING) &&
+        !m_strGenreDescription.empty())
+    {
+      // Type and sub type are both not given. No EPG color coding possible unless sub type is
+      // used to specify EPG_GENRE_USE_STRING leaving type available for genre category, use the
+      // provided genre description for the text.
+      m_genre = Tokenize(m_strGenreDescription);
+    }
+
+    if (m_genre.empty())
+    {
+      // Determine the genre description from the type and subtype IDs.
+      m_genre = StringUtils::Split(
+          CPVREpg::ConvertGenreIdToString(m_iGenreType, m_iGenreSubType),
+          CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_videoItemSeparator);
```

Comment:
Should this use `EPG_STRING_TOKEN_SEPARATOR`?

Reply:
@phunkyfish good look, find too, the `CPVREpgInfoTag::Tokenize` and `CPVREpgInfoTag::DeTokenize` are better about clean code.

Seen this both functions are defined as `static` but only used outside by `CPVREpgDatabase` and `CAddonEpgTag`and there with an class pointer.

Not sure how match in Kodi style, but about performance maybe to use this both as `inline` or like in add-on header with `__forceinline`.

Reply:
Maybe this line is just  bug, sitting there for longer, not introduced by this PR. I only moved the exiting code, if you take a closer look at the commit. I will investigate this.

Reply:
... The line is logically wrong, but does not lead to a bug. If there were multiple genres in the string returned  by `CPVREpg::ConvertGenreIdToString(m_iGenreType, m_iGenreSubType)`, then we must use `EPG_STRING_TOKEN_SEPARATOR`, or better `Tokenize`. But `CPVREpg::ConvertGenreIdToString(m_iGenreType, m_iGenreSubType)` will return exactly one genre, so it currently the delimiter to use is not relevant.

However, I will chnge the code to use `Tokenize`. So it will even work if the implementation of `CPVREpg::ConvertGenreIdToString(m_iGenreType, m_iGenreSubType)` changes to return multiple genres.

Reply:
> seen this both functions are defined as static but only used outside by CPVREpgDatabase and CAddonEpgTagand there with an class pointer.

 Which is absolutely no problem. Member functions not requiring an instance should always be declared static, so they **could** be called even if/when no instance is available.